src/ctl/src/cmd_impl/meta/migration.rs

```diff
@@ -174,18 +174,18 @@ pub async fn migrate(from: EtcdBackend, target: String, force_clean: bool) -> an
     println!("worker nodes migrated");
 
     // catalogs.
-    let databases = PbDatabase::list(&meta_store).await?;
-    let schemas = PbSchema::list(&meta_store).await?;
-    let users = PbUserInfo::list(&meta_store).await?;
-    let tables = PbTable::list(&meta_store).await?;
-    let sources = PbSource::list(&meta_store).await?;
-    let sinks = PbSink::list(&meta_store).await?;
-    let indexes = PbIndex::list(&meta_store).await?;
-    let views = PbView::list(&meta_store).await?;
-    let functions = PbFunction::list(&meta_store).await?;
-    let connections = PbConnection::list(&meta_store).await?;
-    let subscriptions = PbSubscription::list(&meta_store).await?;
-    let secrets = PbSecret::list(&meta_store).await?;
+    let databases = PbDatabase::list_at_snapshot::<EtcdMetaStore>(&snapshot).await?;
```

Comment:
Shall we change the signature to

```rust
async fn list_at_snapshot<S>(snapshot: &S) -> MetadataModelResult<Vec<Self>>
    where
        S: Snapshot,
```

...so that we don't need to specify `::<XxxMetaStore>`.